File: python/Film.py
# ----------------------------------------------------
# CLASS FILM
# Definisi kelas objek film dengan atribut privat (_)
# ----------------------------------------------------
class Film:
    def __init__(self, id_film: str, judul: str, genre: str, durasi: int):
        # Menggunakan underscore (_) untuk menandakan atribut privat (konvensi enkapsulasi python)
        self._id = str(id_film)
        self._judul = str(judul)
        self._genre = str(genre)
        self._durasi = int(durasi)

    # Getter dan Setter untuk atribut privat
    def getId(self) -> str:
        return self._id
    def setId(self, id_film: str) -> None:
        self._id = str(id_film)

    def getJudul(self) -> str:
        return self._judul
    def setJudul(self, judul: str) -> None:
        self._judul = str(judul)

    def getGenre(self) -> str:
        return self._genre
    def setGenre(self, genre: str) -> None:
        self._genre = str(genre)

    def getDurasi(self) -> int:
        return self._durasi
    def setDurasi(self, durasi: int) -> None:
        self._durasi = int(durasi)

    # Method perilaku untuk menampilkan informasi film
    def tampilkan_info(self):
        print(f"ID: {self._id} | Judul: {self._judul} | Genre: {self._genre} | Durasi: {self._durasi} menit")

File: php/Film.php
<?php
// ----------------------------------------------------
// CLASS FILM
// Struktur class PHP lengkap dengan enkapsulasi (private properties)
// dan tambahan atribut path gambar lokal
// ----------------------------------------------------
class Film {
    private string $id;
    private string $judul;
    private string $genre;
    private int $durasi;
    private string $gambar; // Atribut wajib: path file lokal (misal: assets/poster.jpg)

    // Konstruktor kelas Film
    public function __construct(string $id, string $judul, string $genre, int $durasi, string $gambar) {
        $this->id = $id;
        $this->judul = $judul;
        $this->genre = $genre;
        $this->durasi = $durasi;
        $this->gambar = $gambar;
    }

    // Getter dan Setter untuk enkapsulasi data
    public function getId(): string { 
        return $this->id; 
    }
    public function setId(string $id): void { 
        $this->id = $id; 
    }

    public function getJudul(): string { 
        return $this->judul; 
    }
    public function setJudul(string $judul): void { 
        $this->judul = $judul;
    }

    public function getGenre(): string { 
        return $this->genre; 
    }
    public function setGenre(string $genre): void { 
        $this->genre = $genre; 
    }

    public function getDurasi(): int { 
        return $this->durasi; 
    }
    public function setDurasi(int $durasi): void { 
        $this->durasi = $durasi; 
    }

    public function getGambar(): string { 
        return $this->gambar; 
    }
    public function setGambar(string $gambar): void { 
        $this->gambar = $gambar; 
    }
}
